libcloud/compute/drivers/dimensiondata.py

```diff
@@ -422,11 +616,190 @@ def ex_list_vlans(self, location=None, network_domain=None):
+        ET.SubElement(create_node, "ipVersion").text = rule.ip_version
+        ET.SubElement(create_node, "protocol").text = rule.protocol
+        # Setup source port rule
+        source = ET.SubElement(create_node, "source")
+        source_ip = ET.SubElement(source, 'ip')
+        if rule.source.any_ip:
+            source_ip.set('address', 'ANY')
+        else:
+            source_ip.set('address', rule.source.ip_address)
+            source_ip.set('prefixSize', rule.source.ip_prefix_size)
+            source_port = ET.SubElement(source, 'port')
+            source_port.set('begin', rule.source.port_begin)
+            if rule.source.port_end is not None:
+                source_port.set('end', rule.source.port_end)
+        # Setup destination port rule
+        dest = ET.SubElement(create_node, "destination")
+        dest_ip = ET.SubElement(dest, 'ip')
+        if rule.destination.any_ip:
+            dest_ip.set('address', 'ANY')
+        else:
+            dest_ip.set('address', rule.destination.ip_address)
+            dest_ip.set('prefixSize', rule.destination.ip_prefix_size)
+            dest_port = ET.SubElement(dest, 'port')
+            dest_port.set('begin', rule.destination.port_begin)
+            if rule.destination.port_end is not None:
+                dest_port.set('end', rule.destination.port_end)
+        ET.SubElement(create_node, "enabled").text = 'true'
+        placement = ET.SubElement(create_node, "placement")
+        placement.set('position', position)
+
+        response = self.connection.request_with_orgId_api_2(
+            'network/createFirewallRule',
+            method='POST',
+            data=ET.tostring(create_node)).object
+
+        rule_id = None
+        for info in findall(response, 'info', TYPES_URN):
+            if info.get('name') == 'firewallRuleId':
+                rule_id = info.get('value')
+        rule.id = rule_id
+        return rule
+
+    def ex_get_firewall_rule(self, network_domain, rule_id):
+        locations = self.list_locations()
+        rule = self.connection.request_with_orgId_api_2(
+            'network/firewallRule/%s' % rule_id).object
+        return self._to_firewall_rule(rule, locations, network_domain)
+
+    def ex_set_firewall_rule_state(self, rule, state):
+        """
+        Change the state (enabled or disabled) of a rule
+        """
+        update_node = ET.Element('editFirewallrule', {'xmlns': TYPES_URN})
+        update_node.set('id', rule.id)
+        ET.SubElement(update_node, 'enabled').text = str(state).lower()
+        result = self.connection.request_with_orgId_api_2(
+            'network/editFirewallRule',
+            method='POST',
+            data=ET.tostring(update_node)).object
+
+        response_code = findtext(result, 'responseCode', TYPES_URN)
+        return response_code in ['IN_PROGRESS', 'OK']
+
+    def ex_delete_firewall_rule(self, rule):
+        update_node = ET.Element('deleteFirewallrule', {'xmlns': TYPES_URN})
```

Comment:
@tonybaloney I think there is a typo, the XML element name is deleteFirewallRule